File: src/providers/cost.ts
import type { DataProvider, SessionData } from '../types.js';

export interface CostData {
  usd: string | null;
}

function formatUsd(amount: number): string {
  return `$${amount.toFixed(2)}`;
}

export const costProvider: DataProvider = {
  name: 'cost',
  async resolve(session: SessionData): Promise<CostData> {
    const cost = session.cost;
    if (!cost) {
      return { usd: null };
    }

    return {
      usd: formatUsd(cost.total_cost_usd),
    };
  },
};

File: internal/provider/cost.go
package provider

import (
	"fmt"

	"github.com/jheddings/ccglow/internal/types"
)

type costProvider struct{}

func (p *costProvider) Name() string { return "cost" }

func (p *costProvider) Resolve(session *types.SessionData) (*types.ProviderResult, error) {
	cost := map[string]any{
		"usd":   "",
		"total": 0.0,
	}
	if session.Cost != nil {
		cost["usd"] = fmt.Sprintf("$%.2f", session.Cost.TotalCostUSD)
		cost["total"] = session.Cost.TotalCostUSD
	}
	return &types.ProviderResult{
		Values:  map[string]any{"cost": cost},
		Formats: map[string]string{"cost.total": "$%.2f"},
	}, nil
}
